File: src/main/java/io/github/tanguygab/playerlistexpansion/filters/CanSee.java
package io.github.tanguygab.playerlistexpansion.filters;

import org.bukkit.OfflinePlayer;
import org.bukkit.entity.Player;

public class CanSee extends Filter {

    @Override
    public boolean filter(String name, OfflinePlayer viewer) {
        Player player = getOnline(name);
        Player viewerPlayer = viewer.getPlayer();
        return player != null && viewerPlayer != null && viewerPlayer.canSee(player);
    }

}

File: src/main/java/io/github/tanguygab/playerlistexpansion/filters/CanSee.kt
package io.github.tanguygab.playerlistexpansion.filters

import org.bukkit.OfflinePlayer

class CanSee : Filter() {
    override fun filter(name: String?, viewer: OfflinePlayer?): Boolean {
        val player = getOnline(name)
        val viewerPlayer = viewer?.player
        return player != null && viewerPlayer != null && viewerPlayer.canSee(player)
    }
}
